app/dashboard/views.py

```diff
@@ -250,9 +250,19 @@ def receive_tip(request):
 @csrf_exempt
 @ratelimit(key='ip', rate='1/m', method=ratelimit.UNSAFE, block=True)
 def send_tip_2(request):
-    """Handle the second stage of sending a tip."""
-    username = request.session.get('handle')
-    primary_email = request.session.get('email', '')
+    """Handle the second stage of sending a tip.
+
+    TODO:
+        * Convert this view-based logic to a django form.
+
+    Returns:
+        JsonResponse: If submitting tip, return response with success state.
+        TemplateResponse: Render the submission form.
+
+    """
+    from_username = request.session.get('handle', '')
+    primary_from_email = request.session.get('email', '')
```

Comment:
should this be `''`?  a few of the recent production tips have had empty email strings... and i think the default here might be causing that.

Reply:
I just added the default `''` to `primary_from_email` - The only reason I could see the `from_email` being empty is if it was unable to fetch the email address from the session.  I'll add a check in to try and catch that.

Reply:
👍